src/armory/Armory.js

```diff
@@ -121,34 +120,27 @@ class Armory extends React.Component<Props, State> {
 
 	// Gets all of the user's inventory.
 	getUserInventory(): void {
-		const responsePromise = getUser();
-
-		responsePromise.then (
-			response => response.json().then(data => {
-				if(response.status !== 200) {
-					console.log(response.status);
-					toast.error(data.msg);
-					console.log(data);
-				}
-				else {
-					this.setState({
-						inventory: getInventory(data.inventory.tankComponents),
-						chassis: getInventory(data.inventory.tankComponents, 'chassis'),
-						weapons: getInventory(data.inventory.tankComponents, 'weapon'),
-						scanners: getInventory(data.inventory.tankComponents, 'scanner'),
-						scannerAddons: getInventory(data.inventory.tankComponents, 'scannerAddon'),
-						jammers: getInventory(data.inventory.tankComponents, 'jammer'),
-						treads: getInventory(data.inventory.tankComponents, 'treads'),
-						items: getInventory(data.inventory.tankComponents, 'item'),
-					});
-				}
-			})
-		)
+		getUserAPICall(user => {
+			if (user == null) {
```

Comment:
Hmm, this is a big issue. If user does equal null, that means that your flow typing is wrong, and it isn't actually a user. If you look in the getUserAPICall method, it only calls the onLoad with a user, never with null [according to the types at least].